src/soundanalysis.cs

```diff
@@ -0,0 +1,157 @@
+using AVFoundation;
+#if !WATCH
+using CoreMedia;
+#endif
+using CoreML;
+using Foundation;
+using ObjCRuntime;
+
+namespace SoundAnalysis {
+
+	[ErrorDomain ("SNErrorDomain")]
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[Native]
+	enum SNErrorCode : long {
+		UnknownError = 1,
+		OperationFailed,
+		InvalidFormat,
+		InvalidModel,
+	}
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNAudioStreamAnalyzer {
+
+		[DesignatedInitializer]
+		[Export ("initWithFormat:")]
+		IntPtr Constructor (AVAudioFormat format);
+
+		[Export ("addRequest:withObserver:error:")]
+		bool AddRequest (ISNRequest request, ISNResultsObserving observer, [NullAllowed] out NSError error);
+
+		[Export ("removeRequest:")]
+		void RemoveRequest (ISNRequest request);
+
+		[Export ("removeAllRequests")]
+		void RemoveAllRequests ();
+
+		[Export ("analyzeAudioBuffer:atAudioFramePosition:")]
+		void Analyze (AVAudioBuffer audioBuffer, long audioFramePosition);
+
+		[Export ("completeAnalysis")]
+		void CompleteAnalysis ();
+	}
+
+	delegate void SNAudioFileAnalyzerAnalyzeHandler (bool didReachEndOfFile);
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNAudioFileAnalyzer {
+
+		[DesignatedInitializer]
+		[Export ("initWithURL:error:")]
+		IntPtr Constructor (NSUrl url, [NullAllowed] out NSError error);
+
+		[Export ("addRequest:withObserver:error:")]
+		bool AddRequest (ISNRequest request, ISNResultsObserving observer, [NullAllowed] out NSError error);
+
+		[Export ("removeRequest:")]
+		void RemoveRequest (ISNRequest request);
+
+		[Export ("removeAllRequests")]
+		void RemoveAllRequests ();
+
+		[Export ("analyze")]
+		void Analyze ();
+
+		[Async]
+		[Export ("analyzeWithCompletionHandler:")]
+		void Analyze (SNAudioFileAnalyzerAnalyzeHandler completionHandler);
+
+		[Export ("cancelAnalysis")]
+		void CancelAnalysis ();
+	}
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNClassification {
+
+		[Export ("identifier")]
+		string Identifier { get; }
+
+		[Export ("confidence")]
+		double Confidence { get; }
+	}
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNClassificationResult : SNResult {
+
+		[Export ("classifications")]
+		SNClassification [] Classifications { get; }
+#if !WATCH // Remove when CoreMedia is enabled on watchOS
+		[Export ("timeRange")]
+		CMTimeRange TimeRange { get; }
+#endif
+	}
+
+	[Watch (6, 0), TV (13, 0), Mac (10, 15, onlyOn64: true), iOS (13, 0)]
+	[BaseType (typeof (NSObject))]
+	[DisableDefaultCtor]
+	interface SNClassifySoundRequest : SNRequest {
+
+		[Export ("overlapFactor")]
+		double OverlapFactor { get; set; }
+
+		[return: NullAllowed]
```

Comment:
`[return: NullAllowed]` in C# it's not possible to return `null` from a `.ctor` so the attribute is kind of lying (remove it)